### DISCUSSION TITLE
What is variant for in the invariant configurator?

### DISCUSSION THREAD

**Discussion by waynee95** [KeYProject/key#3273] [Q&A] [answered]
Question in title. That field accepts an integer but I am not really sure what it does. I tried to look through the KeY book but haven't found any comment on it.

**Comment by mattulbrich** [comment, marked as answer]
This field allows you to specify a loop variant, an expression whose value (a) never becomes negative and (b) is strictly decreased in every loop iteration. If you can provide such an expression, then termination of the loop is guaranteed.

It is the configurator equivalent of a `decreases` in JML. You can find that in JML or KeY documentation for comparison,

**Reply by waynee95** [reply]
Thank you!
